qa/tasks/mgr/dashboard/test_host.py

```diff
@@ -79,6 +79,27 @@ def test_host_list_with_sources(self):
         resp_hostnames = {host['hostname'] for host in data}
         self.assertEqual(len(test_hostnames.intersection(resp_hostnames)), 0)
 
+    def test_host_list_with_facts(self):
+        # check with facts=true
+        data = self._get('{}?facts=true'.format(self.URL_HOST))
+        self.assertStatus(200)
+        test_hostnames = {inventory_node['name'] for inventory_node in
+                          self.ORCHESTRATOR_TEST_DATA['inventory']}
+        resp_hostnames = {host['hostname'] for host in data}
+        self.assertEqual(test_hostnames, resp_hostnames)
+        for server in data:
+            self.assertIn('cpu_count', server)
```

Comment:
You can create an expected gather-facts info dictionary and use intersection to check if the info is there.